### PR TITLE
Snap progress indicator

### PR DESCRIPTION
**Pull request by d-loose**
Adds a progress indicator for snap operations

[Screencast from 2023-01-31 13:13:51.webm](https://user-images.githubusercontent.com/113362648/215757099-02af8531-1d8f-457e-afbc-ffa15ba882fd.webm)

I'll add something similar for packagekit in a separate PR.

Ref #906

### CODE REVIEW

File: lib/snapd_change_x.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import 'package:snapd/snapd.dart';
+
+extension SnapdChangeX on SnapdChange {
+  double get progress {
```
Comment (Feichtmeier) [inline lib/snapd_change_x.dart:4]: ahh =) this was the part that I had missing!

File: lib/app/common/packagekit/package_page.dart [hunk 2]
```diff
@@ -176,11 +176,6 @@ class _PackagePageState extends State<PackagePage> {
           );
 
     var controls = PackageControls(
```
Comment (Feichtmeier) [inline lib/app/common/packagekit/package_page.dart:178]: why did you move this to a model call again? Just curious. Seems unrelated to me for this PR :thinking:

File: lib/snapd_change_x.dart (hunk 1 above)
Comment (Feichtmeier) [reply lib/snapd_change_x.dart:4]: it's a good thing, works nicely!

File: lib/app/common/packagekit/package_page.dart (hunk 2 above)
Comment (d-loose) [reply lib/app/common/packagekit/package_page.dart:178]: Ah sorry, that's part of my packagekit experimentation - I'll exclude it from here :)

**Review by Feichtmeier — APPROVED** [review]
TY!
